updated remaining source and target types (#600)
SimpleDataLabsInc/prophecy-docs: docs/Spark/gems/source-target/catalog-table/iceberg.md

  livy="Not Supported"
/>

Reads from and writes to Iceberg tables, including Iceberg merge operations and time travel.

## Required Settings

To use the Source gem with the Iceberg catalog table type, you must configure the following required settings at the environment, initialization, and runtime stages.

### Environment Setting

Configure the following Spark JAR dependency package in your fabric environment:

`https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-spark-runtime-3.3_2.12/1.5.0/iceberg-spark-runtime-3.3_2.12-1.5.0.jar`

This dependency is available on your compute platform where you installed Spark, such as your Databricks cluster, EMR, or Dataproc.

### Initialization Settings

Configure the following Spark session property during the Spark session initialization.

- Key: `spark.sql.extensions`
- Value: `org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions`

:::tip
You can also do this during cluster bootstrap. For example, you can set `--properties "spark:spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtentions" \` with a create clusters command.
:::

### Runtime Settings

> The following properties allow you to configure multiple catalogs and your respective metastores for Iceberg tables and data management.

Reviewer: I enjoy this change 😄

| Description       | Description of your dataset.                                                                                                                                          | None    |
| Write Mode        | How to handle existing data. For a list of the possible values, see [Supported write modes](#supported-write-modes).                                                  | None    |
| Merge schema      | Whether to automatically add columns that are present in the `DataFrame` but not in the Target table to the end of the schema as part of a write transaction.         | false   |
| Partition Columns | List of columns to partition the Iceberg table by. <br/>Provide this during a `createOrReplace` write mode to leverage the`overwritePartitions` write mode in future. | None    |

### Supported write modes

| Write Mode          | Description                                                                                             |
| ------------------- | ------------------------------------------------------------------------------------------------------- |

> | createOrReplace     | Create, or replace the Iceberg table.                                                                   |

Reviewer: nit
```suggestion
| createOrReplace     | Create or replace the Iceberg table.                                                                   |
```

| append              | If data already exists, append the contents of the DataFrame to the Iceberg table.                      |
| overwritePartitions | Dynamically overwrite partitions in the Iceberg table.                                                  |
| overwrite           | If data already exists, explicitly overwrite the partitions with the `Overwrite Condition` you specify. |

:::tip

> Among these write modes overwrite and append works the same way as in case of parquet file writes.

Reviewer: ```suggestion
Among these write modes, overwrite and append works the same way for parquet file writes.
```
